plugins/chat/lib/chat_message_creator.rb

```diff
@@ -81,6 +90,52 @@ def validate_channel_status!
     end
   end
 
+  def validate_reply_chain!
+    return if @in_reply_to_id.blank?
+
+    @root_message_id = DB.query_single(<<~SQL).last
+      WITH RECURSIVE root_message_finder( id, in_reply_to_id )
+      AS (
+        -- start with the message id we want to find the parents of
+        SELECT id, in_reply_to_id
+        FROM chat_messages
+        WHERE id = #{@in_reply_to_id}
+
+        UNION ALL
+
+        -- get all parents of the message
```

Comment:
For clarity here, we actually only get the direct parent here, not all the parents right?

I mean this specific block of 3 lines of SQL looks the parent message of a message. The comment can make it think a message can have multiple direct parents.

Reply:
Yes I will update the comment, only the direct parent, though later on if we want a query to get the entire chain of parents we can just remove `WHERE in_reply_to_id IS NULL;` off of this query.